tests/Doctrine/Tests/ORM/Functional/OidReuseTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional;
+
+
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class OidReuseTest extends OrmFunctionalTestCase
+{
+
+    private $userId;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        $this->useModelSet('cms');
+        parent::setUp();
+    }
+
+    public function testOidReuse()
+    {
+        $user = new CmsUser();
+        $this->_em->merge($user);
+
+        $user = null;
```

Comment:
Can you add an assertion that verifies that the spl object hash of `$user` is NOT in the UoW at this point in time?